### PR TITLE
Add doc about collecting trace with assembly loading information

### PR DESCRIPTION
**Pull request by elinor-fung**
## Summary

Add documentation about collecting a trace with detailed assembly loading information

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]
Great writeup!

**Review by vitek-karas — APPROVED** [review]

**Review by IEvangelist — CHANGES_REQUESTED** [review]
We need to add these to the TOC. See https://github.com/dotnet/docs/blob/master/docs/fundamentals/toc.yml

File: docs/core/dependency-loading/collect-details.md
Comment (adegeo) [inline docs/core/dependency-loading/collect-details.md:74]: Please review all of your images. You should use the new `:::image` syntax. This automatically adds a grey border around the image as required. Also, make sure your images are <= 1200 pixels. If you can't fit the content in that size, try using a 100% zoom level on your desktop.
```suggestion
:::image type="content" source="media/collect-details/assembly-loader-filter.png" alt-text="PerfView assembly loader filter image.":::
```

File: docs/core/dependency-loading/collect-details.md
Comment (adegeo) [inline docs/core/dependency-loading/collect-details.md:156]: There are a few references to file/folder paths in the articles. File paths should be enclosed in _italics_

**Review by adegeo — CHANGES_REQUESTED** [review]
Thank you for the articles! This content will be really helpful to our users. I have a few things that need to be addressed before we can move forward.

File: docs/core/dependency-loading/assembly-loader-events.md
Comment (elinor-fung) [reply docs/core/dependency-loading/assembly-loader-events.md:2]: Yes, it is related to diagnostics. The events would go through EventPipe, ETW, and LTTng. I had added this page here with the intent of removing it once we had a place with all the runtime events. Since #20817 is already out for adding that, I'll remove this page and hold off on this change so that we can just link to the content from #20817 once it is in.

**Comment by elinor-fung** [comment]
I fixed the images and updated this change to point to the new docs about runtime events that were just merged. This should be ready for another review now.

@adegeo @IEvangelist Could you take another look?

**Review by IEvangelist — APPROVED** [review]
LGTM - I'll let @adegeo approve it from here and merge. Thanks

**Review by adegeo — APPROVED** [review]
Sorry for the delay!

**Pull request merged**
